src/Provider/AbstractProvider.php

```diff
@@ -638,10 +638,10 @@ protected function getDefaultHeaders($token = null)
      *
      * No default is provided, providers must overload this method to activate
      * authorization headers.
-     *
+     * @param AccessToken|null $token
      * @return array
      */
-    protected function getAuthorizationHeaders($token = null)
+    protected function getAuthorizationHeaders(AccessToken $token = null)
```

Comment:
For instance, storing *only* the access token field of a token, then using that to create a request later?